### DISCUSSION TITLE
国密 TLS

### DISCUSSION THREAD

**Discussion by xuyang2** [emmansun/gmsm#29] [General]
@emmansun 
国密 TLS ，我基于 [github.com/tjfoc/gmsm](https://github.com/tjfoc/gmsm) 的代码，
换成用本项目的 sm2 sm3 sm4 smx509 包
https://github.com/easyops-cn/ntls-go/issues/11 https://github.com/easyops-cn/ntls-go/pull/14
作为国密 TLS 客户端 应该能凑合用了。
（之所以 import tjfoc/gmsm 而不是 fork 是因为其已经不怎么活跃了）

后面其他人如果想做国密 TLS  的话，或许可以参考一下

**Comment by emmansun** [comment]
希望国密开源生态越来越好！
